FWCore/src/components/PileupOverlayAlg.h

```diff
@@ -0,0 +1,69 @@
+#ifndef FWCORE_PILEUPOVERLAYALG_H
+#define FWCORE_PILEUPOVERLAYALG_H
+
+#include "FWCore/DataHandle.h"
+#include "FWCore/IEDMMergeTool.h"
+#include "FWCore/IPileUpTool.h"
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "GaudiKernel/RndmGenerators.h"
+
+#include "podio/EventStore.h"
+#include "podio/ROOTReader.h"
+
+// forward declarations
+class IEDMMergeTool;
+
+
+/*** @class PileupOverlayAlg
+ *  Algorithm for Pileup I/O and merging
+ *
+ * The main purpose of this class is to keep a list of merge tools corresponding to different
+ * datatypes, open a file and let the tools read pileup data and merge it with the signal.
+ * It is thus flexible with regards to the data collection types, as everything is delegated to
+ * tools with a very general interface. One thing this algorithm does do is keep track of the
+ * number of pileup events (via the pileup tool) and the current position in the minimum bias pool
+ * ( which can be randomized ).
+ *
```

Comment:
I think we had a discussion that this information is already in the commits, so I would just leave the author tag out except when the commits are lost.